File: scripts/talker.cpp
#include <time.h>
#include "ros/ros.h"
#include "frequency_test/Freq.h"
#include "frequency_test/SuperAwesome.h"

int main(int argc, char **argv)
{
  ros::init(argc, argv, "talker");
  ros::NodeHandle n;

  ros::Publisher chatter_pub = n.advertise<frequency_test::SuperAwesome>("chatter", 1000);

  
  double start_time = ros::Time::now().toSec();

  int count = 0;
  while (ros::ok())
  {
    double current_time = ros::Time::now().toSec();
    double dt = current_time - start_time;
    double publish_rate = dt * 200 + 1;
    ros::Rate loop_rate(publish_rate);
    
    frequency_test::SuperAwesome message;
    message.data = "Super";
	message.count = count;


    chatter_pub.publish(message);

    ros::spinOnce();

    loop_rate.sleep();

    count++;
  }
	

  return 0;
}

File: scripts/talker.py
#!/usr/bin/env python

import rospy
import time
from std_msgs.msg import String
from frequency_test.msg import SuperAwesome

def talker():
    global start_time
    pub = rospy.Publisher('chatter', SuperAwesome, queue_size=10)
    rospy.init_node('talker', anonymous=True)

    # 'count' stores the total number of message generated and sent via talker.py
    count = 0
    while not rospy.is_shutdown():
        # customize publish rate (linear to time)
        publish_rate = (time.time() - start_time) * 200 + 1
        rate = rospy.Rate(publish_rate)

        message = SuperAwesome()
        message.data = "Super"
        message.count = count
        
        pub.publish(message)
        count += 1
        rate.sleep()

if __name__ == '__main__':
    start_time = time.time()
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

